### ISSUE TITLE
Clarification question: on the VOGN implementation

### ISSUE THREAD

**Issue by PraljakReps** [team-approx-bayes/dl-with-bayes#4]
Amazing repository and work!  

I have a quick clarification question on the VOGN implementation; does the neural network architectures implement reparametrized or Flipout layers? From the repository containing model scripts, this doesn't seem like the case, and regular torch layers are implemented (e.g., dense, conv2d) -- correct me if I am wrong. Thanks for your time!

**Comment by kazukiosawa** [comment]
Thank you for your question!

Yes, you are right. Our implementation supports only `torch.nn.Linear`, `torch.nn.Conv2d`, and `torch.nn.BatchNorm1d/2d`.

**Comment by PraljakReps** [comment]
Another quick question: using this VOGN optimizer, how do you sample from a posterior, or more importantly, measure uncertainty if the model graph does not necessarily implement an explicit distribution over the weights (e.g., reparametrized layers or Flipout)? 

Thanks for your time!

**Comment by kazukiosawa** [comment]
Sampling is defined in the `VIOptimizer` and the child classes of `Curvature` class (e.g. `DiagCurvature`) of [the internal library](https://github.com/cybertronai/pytorch-sso).
See
https://github.com/cybertronai/pytorch-sso/blob/master/torchsso/optim/vi.py#L129-L139
and 
https://github.com/cybertronai/pytorch-sso/blob/master/torchsso/curv/curvature.py#L243-L246.

For modules other than `torch.nn.Linear`, `torch.nn.Conv2d`, and `torch.nn.BatchNorm1d/2d`, no sampling will be done. (see https://github.com/cybertronai/pytorch-sso/blob/master/torchsso/optim/vi.py#L138-L139.) 

[The toy example](https://github.com/team-approx-bayes/dl-with-bayes/tree/master/toy_example) might help to understand how to visualize the entropy of the model prediction w/ and w/o samplings.
Also, we used [the method of scikit-learn](https://scikit-learn.org/stable/modules/generated/sklearn.calibration.calibration_curve.html) to make the calibration curve.

I hope this answers your question!
